custom/enikshay/two_b_release_1/management/commands/enikshay_2b_cbnaat_fix.py

```diff
@@ -0,0 +1,117 @@
+import datetime
+import logging
+
+from corehq.apps.es import CaseSearchES
+from django.core.management import BaseCommand
+from casexml.apps.case.mock import CaseStructure, CaseFactory
+from casexml.apps.case.xform import get_case_updates
+
+logger = logging.getLogger('enikshay_2b_cbnaat_fix')
+
+
+class Command(BaseCommand):
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'domain',
+            help="The domain to migrate."
+        )
+        parser.add_argument(
+            "log_path",
+            help="Path to write the log to"
+        )
+        parser.add_argument(
+            '--commit',
+            action='store_true',
+            help="actually modifies the cases. Without this flag, it's a dry run."
+        )
+
+    @staticmethod
+    def _get_result_recorded_form(test):
+        """get last form that set result_recorded to yes"""
+        for action in reversed(test.actions):
+            for update in get_case_updates(action.form):
+                if update.id == test.case_id:
+                    case_properties = update.get_update_action().dynamic_properties
+                    if case_properties.get('result_recorded') == 'yes':
+                        return action.form.form_data
```

Comment:
This needs a modification I just added to the original: https://github.com/dimagi/commcare-hq/pull/17514/files